src/docfx/config/ops/OpsPreProcessor.cs

```diff
@@ -60,14 +67,39 @@ private void PreProcessMonoDocXml()
                             fallbackOutputDirectory: fallbackOutputDirectory,
                             logWriter: LogError,
                             logContentBaseDirectory: _buildOptions.DocsetPath,
-                            sourceMapFilePath: Path.Combine(_buildOptions.DocsetPath, $".sourcemap-{index}.json"),
+                            sourceMapFilePath: Path.Combine(_buildOptions.DocsetPath, $".sourcemap-ecma-{index}.json"),
                             config: monodocConfig);
                     }
                 }
             }
         }
 
-        private void LogError(LogItem item)
+        private void PreProcessMAML()
+        {
+            if (_config.MAML2YamlMonikerMappingPath is null)
+            {
+                return;
+            }
+
+            using (Progress.Start("Preprocessing MAML markdown files"))
+            {
+                lock (s_maml)
+                {
+                    for (var index = 0; index < _config.MAML2YamlMonikerMappingPath.Length; index++)
```

Comment:
why not just `foreach` on `_config.MAML2YamlMonikerMappingPath`? same in line 47

Reply:
need a `index` number for `sourcemap-{index}`